util/helm/helm.go

```diff
@@ -67,10 +68,26 @@ func (h *helm) Template(templateOpts *TemplateOpts) (string, error) {
 
 func (h *helm) DependencyBuild() error {
 	for _, repo := range h.repos {
-		_, err := h.cmd.RepoAdd(repo.Name, repo.Repo, repo.Creds)
+		if repo.EnableOci {
+			h.cmd.IsHelmOci = true
+			_, err := h.cmd.Login(repo.Repo, repo.Creds)
+			h.cmd.IsHelmOci = false
+
+			defer func() {
+				h.cmd.IsHelmOci = true
+				_, _ = h.cmd.Logout(repo.Repo, repo.Creds)
+				h.cmd.IsHelmOci = false
+			}()
+
+			if err != nil {
+				return err
+			}
+		} else {
```

Comment:
Why is `cmd.RepoAdd` skipped for OCI repositories? I think it still has to be done repo OCI repo can be accessed  via alias, is not it?

Reply:
This is my understanding of helm's current support for OCI registries:

OCI registries do not have an alias. The URL of the registry/repo is what is used to reference charts when pushing/pulling from an OCI registry. 

The only `helm` CLI sub commands that support communicating with an OCI registry are:
`helm registry (login|logout)`
`helm chart (push|pull)`
`helm dependency (build|list|update)`

Running `helm registry login` makes the repo available locally (authenticated) in the same way `helm repo add --username --password` does for non-OCI repos.

Any charts that have a dependency on an OCI repo need to reference the subchart correctly by [prefixing the chart ref ](https://github.com/helm/helm/blob/213a7df2dcd07e8d49fc6acf8a53a540c661ed53/pkg/downloader/manager.go#L398)with `oci://`. Which in turn triggers helm to [use the registry client](https://github.com/helm/helm/blob/213a7df2dcd07e8d49fc6acf8a53a540c661ed53/pkg/action/pull.go#L92) when pulling charts

Reply:
thank you for explanation!